File: usecases/create_todo.py
# usecases/create_todo.py
from entities.todo import Todo
from interfaces.todo_repository import TodoRepositoryInterface

class CreateTodoUseCase:
    def __init__(self, repo: TodoRepositoryInterface):
        self.repo = repo

    def execute(self, title: str) -> Todo:
        todo = Todo(id=self.repo.next_id(), title=title)
        self.repo.save(todo)
        return todo

File: aaa-practic/v6/use_cases/create_todo.rb
require_relative '../entities/todo'

class CreateTodo
  def initialize(repository)
    @repository = repository
  end
  
  def execute(title)
    raise ArgumentError, "Title cannot be empty" if title.nil? || title.strip.empty?
    
    todo_id = @repository.next_id
    todo = Todo.new(id: todo_id, title: title.strip)
    @repository.add(todo)
    todo
  end
end
